### ISSUE TITLE
Calendar Feature to track upcoming events

### ISSUE THREAD

**Issue by campgurus** [BOWiki/BOW#381]
Seeing this: http://homicidewatch.org/calendar/ reminded me of something we had spoken about a while back. Our goals are Information, Organization and Action. This feature falls within the Action category. It looks like there are gems that do this like this one: https://github.com/excid3/simple_calendar. 

Or if we wanted to roll our own from scratch, here is a railscast: https://www.youtube.com/watch?v=ndBKSjAbemo or this : https://richonrails.com/articles/building-a-basic-calendar-in-ruby-on-rails.

Specifically, a User should be able to:
* Post events relating to a case or cases Event has_and_belongs_to_many cases (?)
* Edit events they have created
* View a month at a glance calendar view of links to event details
* View events by geography
* Toggle between list of events and calendar view

Events could include court dates, protests and possibly other types of events. Events will be listed by case name on the calendar and potentially use tokenized colored buttons to distinguish between the different types of events. Events could be shareable on Twitter.

**Comment by rlgreen91** [comment]
Where are we with implementing this?

**Comment by campgurus** [comment]
though i think we agreed that this feature is an important part of what we are building, i believe we had tabled it in favor of some other priorities related to our tech debt. We may be closer now to trying this.

**Comment by rlgreen91** [comment]
Right, I get shifting priorities, but let's not leave stale branches of things that could be deployed.  Is this ready to be submitted as a PR?

**Comment by campgurus** [comment]
This branch is gone, with my stolen computer, unfortunately.  If we want this feature, it was not difficult to implement.  Just need to create a Calendar events model to populate it.

If i recall correctly, the only issue I had with implementation was  that The calendar was not centered on the page.

I wouldn't mind putting this feature together again and putting in a PR over the weekend.

**Comment by trystant** [comment]
I think we should prioritize this and tackle it as it comes up in the queue. People have historically jumped on tickets as they felt the need or desire, but I feel that we want to move differently moving forward.

**Comment by campgurus** [comment]
@trystant i agree we (mostly "I") should be more organized about which issue to tackle next. It would be great to get these issues organized in a queue or at least grouped into priorities, so when one of us has time, we can pick them off in a more organized and predictable way. 

I know we were moving away from the P labels.  How do we now identify the order of priorities?

**Comment by campgurus** [comment]
https://www.driftingruby.com/episodes/fullcalendar-events-and-scheduling

**Comment by campgurus** [comment]
for implementing follow relationships with events.  this seems like the easiest way -- using separate controllers for each followable. if not as elegant as a polymorphic relationship:

https://stackoverflow.com/questions/20108709/acts-as-follower-for-multiple-models

**Comment by rlgreen91** [comment]
If we implement this, then it would also be good to, in the parent follows controller, throw an error if the `followable` method is not implemented in the child controller.

**Comment by campgurus** [comment]
Here is a question: fullcalendar.js provides the calendar view with a month at a time, week or just one day.  Alternatively, we could just have a simple list of events.  Which is better?

**Comment by rlgreen91** [comment]
I definitely assumed that you included a calendar view in the branch.

So here's my thoughts: this is a jQuery plugin.  Do we already use jQuery in our app?  How do we actually include this?  Are we having to use a build tool like Grunt or Gulp to properly include everything?  Because that will introduce complexity not just in staging and production but also in setting up your local environment.

**Comment by campgurus** [comment]
we incorporate the calendar view via the gem..  the calendar view is included in the current branch. thinking about it now, the view is not a problem, and obviates some of the need for adding a search feature, since people can at least browse by date by going to the appropriate place on the calendar. I was just trying to figure out if we could find a way to simplify the feature so that we can add it.

**Comment by rlgreen91** [comment]
Well, we really would still want to have a search feature.  You many not know when an event is taking place, want to find events located near you, that type of thing.

**Comment by campgurus** [comment]
I agree. I was definitely planning to incorporate search.  I was just trying to get a minimum events feature working, where users could at least view uploaded events on a calendar at first, and then when that is solid, adding ability for users to search as well as other features listed in the initial issue description above.

**Comment by trystant** [comment]
I think that the phased approach makes sense with one view as a deliverable but a problem with the short cuts that get us to a goal is that the work around the short cut often takes a long time to fix later.  Also, jquery is specified in `app/assets/javascripts/application.js`
